### PR TITLE
Redesign action icons

### PR DESCRIPTION
**Pull request by rolandlo**
This is #2886 squashed and rebased, introducing the new action icons by @Fatih20 with some accompanying code changes by @Technius and myself. Huge thanks to @Fatih20 for contributing the icons!

Fatih20's changes:
* All action icons revamped
* Renamed some icons, making the naming scheme more systematic
* Added new icons for layer combo, drawing type combo, selection combo and zoom slider

Technius's changes:
* Adjusted icon load paths

rolandlo's changes:
* Added option to use available stock icons
* Prefixed all icons with "xopp-"

### CODE REVIEW

**Review by Technius — APPROVED** [review]
Looks fine to me.
* The `IconNameHelper` change seems reasonable to me since it needs to depend on `Settings`.
* Can you split into a single separate commit: renaming of `Layer selection` to `Layer Combo` and fixing capitalization of `Zoom Slider`.
* There's a lot of extraneous formatting changes that I think would be better in a separate commit (e.g. `XournalMain.cpp` line 286).

I'm happy that we've gone this far. Thanks for all the hard work @rolandlo and @Fatih20!

**Comment by rolandlo** [comment]
>     * Can you split into a single separate commit: renaming of `Layer selection` to `Layer Combo` and fixing capitalization of `Zoom Slider`.
> 
>     * There's a lot of extraneous formatting changes that I think would be better in a separate commit (e.g. `XournalMain.cpp` line 286).
> 

Ok, done!

**Comment by rolandlo** [comment]
Merging in 24 hours if no objections are raised

**Comment by mjg** [comment]
Merges and builds cleanly (locally) and looks good so far. Thanks :)

File: src/control/settings/SettingsEnums.h [hunk 1]
```diff
@@ -11,6 +11,7 @@
 
 #pragma once
 
+#include <optional>
```
Comment (Febbe) [inline src/control/settings/SettingsEnums.h:14]: Does not belong here I think

File: src/gui/dialog/ButtonConfigGui.cpp [hunk 2]
```diff
@@ -10,7 +10,7 @@
 #include "Util.h"
 #include "i18n.h"
 
-#define ADD_TYPE_CB(icon, name, action) addToolToList(typeModel, icon, name, action)
+#define ADD_TYPE_CB(icon, name, action) addToolToList(typeModel, iconNameHelper.iconName(icon).c_str(), name, action)
```
Comment (Febbe) [inline src/gui/dialog/ButtonConfigGui.cpp:13]: I would use a lambda instead.

File: src/gui/toolbarMenubar/ToolMenuHandler.cpp [hunk 3]
```diff
@@ -309,12 +300,18 @@ void ToolMenuHandler::signalConnectCallback(GtkBuilder* builder, GObject* object
     addToolItem(new ToolButton(listener, name, action, stockIcon, text))
 
 // Use Custom loading Icon
-#define ADD_CUSTOM_ITEM(name, action, icon, text) addToolItem(new ToolButton(listener, name, action, icon, text))
+#define ADD_CUSTOM_ITEM(name, action, icon, text) \
```
Comment (Febbe) [inline src/gui/toolbarMenubar/ToolMenuHandler.cpp:303]: Maybe also a lambda for the following `#define`s

File: src/gui/dialog/ButtonConfigGui.cpp (hunk 2 above)
Comment (rolandlo) [reply src/gui/dialog/ButtonConfigGui.cpp]: Done. Please check!

File: src/gui/toolbarMenubar/ToolMenuHandler.cpp (hunk 3 above)
Comment (rolandlo) [reply src/gui/toolbarMenubar/ToolMenuHandler.cpp]: Done Please check!

File: src/control/settings/SettingsEnums.h (hunk 1 above)
Comment (rolandlo) [reply src/control/settings/SettingsEnums.h]: Done!

**Comment by rolandlo** [comment]
I have checked the Windows installer (built on Azure pipelines). Icons work just fine. No problem at all. When installing without first uninstalling the previous version, the user is left with old icon folders, but they do not disturb. The 1.1.0+dev uninstaller removes all icon folders, so after uninstalling and freshly installing only the new icon folders are there.

File: src/gui/toolbarMenubar/ToolMenuHandler.cpp (hunk 3 above)
Comment (Febbe) [reply src/gui/toolbarMenubar/ToolMenuHandler.cpp]: LGTM.

**Comment by Febbe** [comment]
I think there is no one left with objections, so you can safely merge it.

**Comment by rolandlo** [comment]
> I think there is no one left with objections, so you can safely merge it.

Great! By the way, I have also checked if the icons work flawlessly on MacOS. They do.

**Comment by rolandlo** [comment]
Again, huge thanks to you, @Fatih20 . You did a wonderful job and have invested countless hours. I am very happy we finally have the new icons and got a huge step closer to releasing version 1.1.0. 
@idotobi Please update the copyrights PR #3120

**Comment by idotobi** [comment]
@rolandlo : Will do so this weekend.

@Fatih20 : Thank you for the great looking icons. Looking forward to using them :nerd_face:

**Comment by Fatih20** [comment]
Also, you guys might want to update the screenshot in the github page to the one using the icon once the update is live.

**Comment by rolandlo** [comment]
> Also, you guys might want to update the screenshot in the github page to the one using the icon once the update is live.

Yes, the screenshots for the Readme shown on the github main page will get replaced with screenshots showing the new icons. Some more updates will have to be made for the Readme, as for instance MacOS HighSierra is not supported any more, but MacOS BigSur and MacOS Catalina are. Also the [website](https://xournalpp.github.io/) should get some screenshots, that will use the new icons.

**Pull request merged**